### PR TITLE
functions/package: add a comment/description to pkg_compare_version funciton

### PR DESCRIPTION
**Pull request by jouvin**
Syntax of the annotation incorrect, I have a meeting, I'll tro fix it when I am done with it.

### CODE REVIEW

File: quattor/functions/package.pan [hunk 1]
```diff
@@ -1,6 +1,13 @@
 declaration template quattor/functions/package;
 
-# Compare package versions
+@{
+desc = function to compare 2 package versions in the form x.y.z-r. It properly handles \
+ number comparison of each element (10 is considered greater than 2) and it also accepts \
+ letters in one of the element (a string comparison is done for the element in this case).
+args = the two version strings to compare.
+return value = -1 if the second argument is greater than the first one, 0 if they are equal \
```
Comment (jrha) [inline quattor/functions/package.pan]: -1 if the second argument is less than the first, 0 if they are equal

File: quattor/functions/package.pan [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+ and 1 if the first argument is greated than the second one.
```
Comment (jrha) [inline quattor/functions/package.pan]: and 1 if the second argument is greater than the first.

**Comment by jrha** [comment]
I was wondering whether we should add something like:

```pan
final variable PKG_VERSION_GREATER = 1;
final variable PKG_VERSION_EQUAL = 0;
final variable PKG_VERSION_LESS = -1;
```

and then update the function to also use the "constants", it might help avoid confusion.

File: quattor/functions/package.pan (hunk 2 above)
Comment (jouvin) [inline quattor/functions/package.pan]: Good catch! The last commit should have a valid syntax and the right contents!

**Comment by jouvin** [comment]
You are probably right... Feel free to do it, I am not sure if I will find the time today...

**Comment by jouvin** [comment]
The test failure is due to something wrong with `yumng`...

**Comment by jouvin** [comment]
Done

**Comment by jrha** [comment]
Tests are failing because `SOFTWARE_GROUP` is defined separately for both `yum` and `yumng` and the tests just include both schemas, which should not happen in real usage...

**Comment by stdweird** [comment]
shared types should be reused, not redefined imho. i opened https://github.com/quattor/configuration-modules-core/issues/968 to follow up

**Comment by stdweird** [comment]
retest this please

**Comment by jrha** [comment]
Redefinition is the best way to achieve work around lack of sub-classing, in this case it is a non-issue anyway (see quattor/configuration-modules-core#968).

**Pull request merged**